File: python/0010.py
#
# Solution for Project Euler problem 10
# Copyright michimani All rights reserved.
#
# https://projecteuler.net/problem=10
#
import math
UNTIL = 2000000


def compute():
    ans = 2
    num = 3
    while num <= UNTIL:
        is_prime = True
        for dividor in range(3, math.floor(num ** (1/2)) + 1, 2):
            if num % dividor == 0:
                is_prime = False
                break
        if is_prime is True:
            ans = ans + num

        print('{} %'.format(round((num / UNTIL), 2) * 100))
        num = num + 2
    return ans


if __name__ == "__main__":
    print(compute())

File: go/solution/0010.go
// Solution for Project Euler problem 10
// Copyright michimani All rights reserved.
//
// https://projecteuler.net/problem=10
package solution

import (
	"fmt"
	"math"
)

// Solve0010 is function to solve the problem.
func Solve0010() {
	var answer int

	for num := 2; num < 2000000; num++ {
		isPrime := true
		for d := 2; d < int(math.Sqrt(float64(num))+1); d++ {
			if num%d == 0 {
				isPrime = false
				break
			}
		}

		if isPrime {
			answer = answer + num
		}
	}

	fmt.Println(answer)
}
